ui_feature.go

```diff
@@ -303,3 +304,34 @@ func convertDataTableToRuleExceptions(table *godog.Table) map[string]a11y.Rule {
 
 	return rules
 }
+
+func (f *UIFeature) iShouldBeRedirectedTo(expectedURL string) error {
+	err := f.CheckLocationUntilTimeOut(expectedURL)
+	if err != nil {
+		return f.StepError()
+	}
+
+	return f.StepError()
+}
+
+func (f *UIFeature) CheckLocationUntilTimeOut(expectedURL string) error {
+	var actualURL string
+
+	for start := time.Now(); ; {
```

Comment:
This is a situation where you are checking for a condition being met rather than iterating n number of times. So I think it would be better to use a loop that checks for the condition, like this: 
```suggestion
	start := time.Now()
	for time.Since(start) <= f.WaitTimeOut {
		err := chromedp.Run(f.Chrome.Ctx, chromedp.Tasks{
			chromedp.Location(&actualURL),
		})
	
		if err != nil {
			return err
		}
	
		if expectedURL == actualURL {
			return nil
		} 
	
		time.Sleep(50 * time.Millisecond)
	}
	return f.StepError()
```

Reply:
Yes, your suggestion is better I have made changes

Reply:
👍